rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/WorkspaceInfo.rsc

```diff
@@ -71,70 +71,46 @@ alias MayOverloadFun = bool(set[loc] defs, map[loc, Define] defines);
 alias FileRenamesF = rel[loc old, loc new](str newName);
 alias RenameLocation = tuple[loc l, Maybe[ChangeAnnotationId] annotation];
 alias DefsUsesRenames = tuple[set[RenameLocation] defs, set[RenameLocation] uses, FileRenamesF renames];
-alias ProjectFiles = rel[loc projectFolder, loc file, bool loadModel];
-
-/**
- * This is a subset of the fields from analysis::typepal::TModel, specifically tailored to refactorings.
- * WorkspaceInfo comes with a set of functions that allow (incrementally) loading information from multiple TModels, and doing (cached) queries on this data.
- */
-data WorkspaceInfo (
-    // Instance fields
-    // Read-only
-    rel[loc use, loc def] useDef = {},
-    set[Define] defines = {},
-    set[loc] sourceFiles = {},
-    map[loc, Define] definitions = (),
-    map[loc, AType] facts = (),
-    Scopes scopes = (),
-    Paths paths = {},
-    set[loc] projects = {}
-) = workspaceInfo(
-    PathConfig(loc) getPathConfig,
-    ProjectFiles() preloadFiles,
-    ProjectFiles() allFiles,
-    set[TModel](ProjectFiles) tmodelsForLocs
+alias ProjectFiles = rel[loc projectFolder, bool loadModel, loc file];
+
+// Extend the TModel to include some workspace information.
+data TModel (
+    set[loc] projects = {},
+    set[loc] sourceFiles = {}
 );
 
 set[RenameLocation] annotateLocs(set[loc] locs, Maybe[ChangeAnnotationId] annotationId = nothing()) = {<l, annotationId> | l <- locs};
 
-WorkspaceInfo loadLocs(WorkspaceInfo ws, ProjectFiles projectFiles) {
-    for (tm <- ws.tmodelsForLocs(projectFiles)) {
-        ws = loadTModel(ws, tm);
+TModel loadLocs(TModel wsTM, ProjectFiles projectFiles, set[TModel](ProjectFiles projectFiles) tmodelsForFiles) {
+    for (modTM <- tmodelsForFiles(projectFiles)) {
+        wsTM = loadTModel(wsTM, modTM);
     }
 
     // In addition to data from the TModel, we keep track of which projects/modules we loaded.
-    ws.sourceFiles += projectFiles.file;
-    ws.projects += projectFiles.projectFolder;
-
-    return ws;
-}
-
-WorkspaceInfo preLoad(WorkspaceInfo ws) {
-    return loadLocs(ws, ws.preloadFiles());
-}
+    wsTM.sourceFiles += projectFiles.file;
+    wsTM.projects += projectFiles.projectFolder;
 
-WorkspaceInfo loadWorkspace(WorkspaceInfo ws) {
-    return loadLocs(ws, ws.allFiles());
+    return wsTM;
 }
 
-WorkspaceInfo loadTModel(WorkspaceInfo ws, TModel tm) {
+TModel loadTModel(TModel wsTM, TModel modTM) {
```

Comment:
should it be named `mergeTModels` instead of load `TModel`?

Reply:
Good one. Fixed.